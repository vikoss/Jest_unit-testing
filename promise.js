import { get } from 'axios';

const getDataFromAPI = (url) => {
  return get(url)
    .then(destructuredData);
};

const character = () => {
  return get('https://rickandmortyapi.com/api/character/1')
    .then(destructuredData)
};

const destructuredData = ({ data }) => data;

export { getDataFromAPI, character };