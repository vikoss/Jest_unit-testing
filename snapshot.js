export const getCharacter = (data) => ({
  id: data.id,
  nombre: data.nombre,
  genero: data.gender,
});