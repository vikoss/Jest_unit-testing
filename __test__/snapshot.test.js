import { getCharacter } from './../snapshot';
import { character } from './../promise'

describe('Snapshots', () => {
  test('Snapshot character', async () => {
    const response = await character();
    expect(getCharacter(response)).toMatchSnapshot();
  });
  test('Always failure', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      createdAt: new Date(),
    };
    expect(user).toMatchSnapshot();
  });
  test('Exception snapshot', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      nickName: 'Vikoss',
    };
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});