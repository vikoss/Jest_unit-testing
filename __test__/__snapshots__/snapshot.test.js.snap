// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Always failure 1`] = `
Object {
  "createdAt": 2021-08-06T16:24:23.285Z,
  "id": 9,
}
`;

exports[`Snapshots Exception snapshot 1`] = `
Object {
  "id": Any<Number>,
  "nickName": "Vikoss",
}
`;

exports[`Snapshots Snapshot character 1`] = `
Object {
  "genero": "Male",
  "id": 1,
  "nombre": undefined,
}
`;
